hyperspy/axes.py

```diff
@@ -528,6 +587,20 @@ def value2index(self, value, rounding=round):
             elif rounding is math.floor:
                 rounding = np.floor
 
+            if value.dtype.type is np.str_:
+                if value[0][0].isdigit():
+                    try:
+                        value = self._get_value_from_value_with_units(value)
+                    except pint.errors.UndefinedUnitError:
+                        raise ValueError(f"Expected iterable of calibrated value with units like ['20nm'] but '{value}' could not be interpreted this way.")
+                else:
+                    try:
+                        assert value[0].startswith('rel')
+                        relative_value = np.char.replace(value, 'rel', '').astype(float)
+                    except (AssertionError, ValueError):
+                        raise ValueError(f"Expected array of relative units of kind ['rel0.2', 'rel0.3] but '{value}' could not be interpreted this way.")
```

Comment:
Similar as above, it should be possible to refactor to avoid repetition and checking if the input is suitable at the right place.